java/ql/src/semmle/code/java/frameworks/android/SharedPreferences.qll

```diff
@@ -0,0 +1,51 @@
+import java
+
+/** The interface `android.content.SharedPreferences`. */
+class SharedPreferences extends Interface {
+  SharedPreferences() { this.hasQualifiedName("android.content", "SharedPreferences") }
+}
+
+/** The class `androidx.security.crypto.EncryptedSharedPreferences`, which implements `SharedPreferences` with encryption support. */
+class EncryptedSharedPreferences extends Class {
+  EncryptedSharedPreferences() {
+    this.hasQualifiedName("androidx.security.crypto", "EncryptedSharedPreferences")
+  }
+}
+
+/** The `create` method of `androidx.security.crypto.EncryptedSharedPreferences`. */
+class CreateEncryptedSharedPreferencesMethod extends Method {
+  CreateEncryptedSharedPreferencesMethod() {
+    this.getDeclaringType() instanceof EncryptedSharedPreferences and
+    this.hasName("create")
+  }
+}
+
+/** Returns `android.content.SharedPreferences.Editor` from the `edit` call of `android.content.SharedPreferences`. */
+class GetSharedPreferencesEditorMethod extends Method {
+  GetSharedPreferencesEditorMethod() {
+    this.getDeclaringType() instanceof SharedPreferences and
+    this.hasName("edit") and
+    this.getReturnType() instanceof SharedPreferencesEditor
+  }
+}
+
+/** The interface `android.content.SharedPreferences.Editor`. */
+class SharedPreferencesEditor extends Interface {
+  SharedPreferencesEditor() { this.hasQualifiedName("android.content", "SharedPreferences$Editor") }
+}
+
+/** A method that updates a key-value pair in a `android.content.SharedPreferences` through a `SharedPreferences.Editor`. The value is not written until a `StorePreferenceMethod` is called. */
```

Comment:
```suggestion
/** 
 * A method that updates a key-value pair in a
 * `android.content.SharedPreferences` through a `SharedPreferences.Editor`. The
 * value is not written until a `StorePreferenceMethod` is called.
 */
```
This ended up being a bit too long to fit on one line.

Reply:
Changed